jetty-http3/http3-server/src/main/java/org/eclipse/jetty/http3/server/internal/ServerHTTP3Session.java

```diff
@@ -170,21 +176,27 @@ public void onSettings(SettingsFrame frame)
         {
             if (key == SettingsFrame.MAX_TABLE_CAPACITY)
             {
-                int maxTableCapacity = value.intValue();
+                int maxTableCapacity = (int)Math.min(value, Integer.MAX_VALUE);
```

Comment:
You're capping positive values to `Integer.MAX_VALUE`, but can't the long value be negative too, in which case it should be capped or rejected?

Reply:
No the value cannot be negative, as it is encoded as a variable-length integer (https://datatracker.ietf.org/doc/html/rfc9000#name-variable-length-integer-enc).